### PR TITLE
feat: auto batch size supports methods that return a dict

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
- also add missing UT for execute_all

### CODE REVIEW

File: deepmd/pt/utils/auto_batch_size.py [hunk 1]
```diff
@@ -78,26 +104,44 @@ def execute_with_batch_size(
...
+                ret = torch.cat(r, dim=0)
             else:
                 raise RuntimeError(f"Unexpected result type {type(r[0])}")
-        r = tuple(r_list)
-        if len(r) == 1:
-            # avoid returning tuple if callable doesn't return tuple
-            r = r[0]
+            return ret
+
+        if not self.returned_dict:
```
Comment (njzjz) [inline deepmd/pt/utils/auto_batch_size.py:139]: I think we don't need this argument? Just check the type of results

File: deepmd/pt/utils/auto_batch_size.py [hunk 2]
```diff
@@ -12,6 +12,32 @@
...
+
+    Parameters
+    ----------
+    initial_batch_size : int, default: 1024
+        initial batch size (number of total atoms) when DP_INFER_BATCH_SIZE
+        is not set
+    factor : float, default: 2.
+        increased factor
+    returned_dict:
+        if the batched method returns a dict of arrays.
```
Comment (anyangml) [inline deepmd/pt/utils/auto_batch_size.py:25]: bool, default: false

File: deepmd/pt/utils/auto_batch_size.py [hunk 3]
```diff
@@ -78,26 +104,44 @@ def execute_with_batch_size(
...
-            if n_batch:
-                for rr in result:
-                    rr.reshape((n_batch, -1))
-                results.append(result)
-        r_list = []
-        for r in zip(*results):
+
+            def append_to_list(res_list, res):
+                if n_batch:
+                    res_list.append(res)
```
Comment (anyangml) [inline deepmd/pt/utils/auto_batch_size.py:116]: no need to reshape?

File: deepmd/pt/utils/auto_batch_size.py (hunk 1 above)
Comment (wanghan-iapcm) [reply deepmd/pt/utils/auto_batch_size.py]: good idea plz check [3c4ff2c](https://github.com/deepmodeling/deepmd-kit/pull/3626/commits/3c4ff2c1c34675c93b853902fd6b0d699034479a)

File: deepmd/pt/utils/auto_batch_size.py (hunk 2 above)
Comment (wanghan-iapcm) [reply deepmd/pt/utils/auto_batch_size.py]: the argument is removed according to @njzjz 's suggestion.

File: deepmd/pt/utils/auto_batch_size.py (hunk 3 above)
Comment (wanghan-iapcm) [reply deepmd/pt/utils/auto_batch_size.py:116]: no the shape of the returned tensors are preserved. A concatenation is performed on the first (batch size) dimension of batch calculated tensors. plz see 
https://github.com/deepmodeling/deepmd-kit/blob/3c4ff2c1c34675c93b853902fd6b0d699034479a/deepmd/pt/utils/auto_batch_size.py#L132-L139

**Review by njzjz — APPROVED** [review]

**Review by anyangml — APPROVED** [review]

**Pull request merged**